corehq/apps/locations/bulk.py

```diff
@@ -126,16 +127,32 @@ def import_location(self, location_type, location_data, parent_child_map=None):
         parent = parent_id
         if provided_code:
             existing = Location.by_site_code(self.domain, provided_code)
-            if existing:
-                if existing.location_type != location_type:
+        if location_id:
```

Comment:
You can save some db calls if you do:
```
if location_id:
    ...
elif provided_code:
    ...
```
Since you'll never use the `provided_code` lookup if `location_id` is provided (if the id lookup fails, the row fails).